host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1034,42 +1036,105 @@ static FILE *open_filename(RASPIVID_STATE *pState, char *filename)
+               goto _error;
+            }
 
-         unsigned short port;
-         sscanf(colon + 1, "%hu", &port);
-         *colon = 0;
+            int sockListen = socket(AF_INET, SOCK_STREAM, 0);
+            if (sockListen < 0)
+            {
+               fprintf(stderr, "error %d creating socket\n", errno);
+               goto _error;
+            }
+
+            serv_addr.sin_family = AF_INET;
+            serv_addr.sin_port = htons(serv_addr.sin_port);
+            int true = 1;
+            setsockopt(sockListen, SOL_SOCKET, SO_REUSEADDR, &true, sizeof(int));
+            if (bind(sockListen, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0)
+            {
+               close(sockListen);
+               fprintf(stderr, "error %d on bind\n", errno);
+               goto _error;
+            }
+
+            fprintf(stderr, "Waiting for TCP connection on %s:%"SCNu16"...",
+                    inet_ntoa(serv_addr.sin_addr), ntohs(serv_addr.sin_port));
+
+            if (-1 == listen(sockListen, 0))
+            {
+               close(sockListen);
+               fprintf(stderr, "error %d on listen\n", errno);
+               goto _error;
+            }
+            socklen_t clilen = sizeof(cli_addr);
+            sfd = accept(sockListen, (struct sockaddr *) &cli_addr, &clilen);
+            if (sfd < 0)
+            {
+               close(sockListen);
+               fprintf(stderr, "error %d on accept\n", errno);
+               goto _error;
+            }
 
-         fprintf(stderr, "Connecting to %s:%hu\n", filename, port);
-         
-         struct sockaddr_in saddr;
-         saddr.sin_family = AF_INET;
-         saddr.sin_port = htons(port);
-         inet_aton(filename, &saddr.sin_addr);
+            fprintf(stderr, "Client connected from %s:%"SCNu16"\n", inet_ntoa(cli_addr.sin_addr), ntohs(cli_addr.sin_port));
+            close(sockListen); //do not listen on a given port anymore
+         }
+      }
 
-         if(connect(sfd, (struct sockaddr *)&saddr, sizeof(struct sockaddr_in)) < 0)
+      if(network)
+      {
+         if (!network_listen)
          {
-            perror("connect");
+            filename += 6;
+            char *colon;
+            colon = strchr(filename, ':');
+            sfd = socket(AF_INET, socktype, 0);
+            if (sfd < 0)
+            {
+               perror("socket");
+            }
+
+            unsigned short port;
+            sscanf(colon + 1, "%hu", &port);
+            *colon = 0;
+
+            fprintf(stderr, "Connecting to %s:%hu\n", filename, port);
+
+            struct sockaddr_in saddr;
+            saddr.sin_family = AF_INET;
+            saddr.sin_port = htons(port);
+            inet_aton(filename, &saddr.sin_addr);
+
+            if (connect(sfd, (struct sockaddr *) &saddr, sizeof(struct sockaddr_in)) < 0)
+            {
+               perror("connect");
```

Comment:
TEMP_FAILURE_RETRY() is the macro. It's defined in unistd.h if __USE_GNU is defined, which I think we are defining already (via _GNU_SOURCE).

Using it will stop this being easily portable to other glibc versions (e.g. musl) so maybe this should be done in a followup patch?